src/classes/CollisionSolution.ts

```diff
@@ -0,0 +1,203 @@
+  public hitWallTestTank(tank: Tank): boolean {
+    let hitWallTest: boolean = false;
+    if (tank.y <= this.wallList[0] || tank.y >= this.wallList[2] || tank.x <= this.wallList[3] || tank.x >= this.wallList[1]) {
+      hitWallTest = true;
+    }
+
+    return hitWallTest;
+  }
+
+  public hitEnemyTestTank(tank: Tank): boolean {
+    let hitEnemyTest: boolean = false;
+    for (const item of this.tankList) {
+      if (item.x === tank.x && item.y === tank.y) {
+        hitEnemyTest = true;
+      }
+    }
+
+    return hitEnemyTest;
+  }
+
+  public hitBulletTestTank(tank: Tank): boolean {
+    let hitBulletsTest: boolean = false;
+    for (const item of this.bulletList) {
+      if (item.x === tank.x && item.y === tank.y) {
+        hitBulletsTest = true;
+      }
+    }
+
+    return hitBulletsTest;
+  }
+
+  public hitDeathTankTestTank(tank: Tank): boolean {
+    let hitDTanksTest: boolean = false;
+    for (const item of this.tankList) {
+      if (item.health <= 0 && item.x === tank.x && item.y === tank.y) {
+        hitDTanksTest = true;
+      }
+    }
+
+    return hitDTanksTest;
+  }
+
+  public scanTanks(tank: Tank): void {
+    const dirThis: number = (tank.direction) * Math.PI / 180;
+    const coefA: number = CONFIG.smallVisionSide;
+    const coefB: number = CONFIG.bigVisionSide;
+    const xA: number = Math.round(coefA * Math.sin(dirThis) - coefA * Math.cos(dirThis) + tank.x);
+    const yA: number = Math.round(coefA * Math.cos(dirThis) - coefA * Math.sin(dirThis) + tank.y);
+    const xB: number = Math.round(coefB * Math.cos(dirThis) + coefA * Math.sin(dirThis) + tank.x);
+    const yB: number = Math.round(coefB * Math.sin(dirThis) + coefA * Math.cos(dirThis) + tank.y);
+    const xC: number = Math.round(coefB * Math.cos(dirThis) - coefA * Math.sin(dirThis) + tank.x);
+    const yC: number = Math.round(coefB * Math.sin(dirThis) - coefA * Math.cos(dirThis) + tank.y);
+    const xD: number = Math.round(-coefA * Math.sin(dirThis) - coefA * Math.cos(dirThis) + tank.x);
+    const yD: number = Math.round(-coefA * Math.cos(dirThis) - coefA * Math.sin(dirThis) + tank.y);
+
+    const maxX: number = Math.max(xA, xB, xC, xD);
+    const maxY: number = Math.max(yA, yB, yC, yD);
+    const minX: number = Math.min(xA, xB, xC, xD);
+    const minY: number = Math.min(yA, yB, yC, yD);
+
+    for (const item of this.tankList) {
+      tank.enemyTracks = [];
+      if (item.x >= minX && item.x <= maxX && item.y >= minY && item.y <= maxY) {
+        tank.enemyTracks.push({
+          id: item.id,
+          x: item.x,
+          y: item.y,
+          direction: item.direction,
+          health: item.health
+        });
+      }
+    }
+
+  }
+
+  public scanBullets(tank: Tank): void {
+    const dirThis: number = (tank.direction) * Math.PI / 180;
+    const coefA: number = CONFIG.smallVisionSide;
+    const coefB: number = CONFIG.bigVisionSide;
+    const xA: number = Math.round(coefA * Math.sin(dirThis) - coefA * Math.cos(dirThis) + tank.x);
```

Comment:
looks like this method is duplication of scanWalls. If so, then make a separate method

Reply:
No. these methods simply use one algorithm to determine the zone of visibility. Has deduced algorithm of definition of a zone of visibility in a separate method.